### PR TITLE
ERFA Overhaul

### PR DESCRIPTION
**Pull request by helgee**
Okay, I have changed a lot of things. But I split the changes in nice thematic commits...

### CODE REVIEW

**Comment by helgee** [comment]
CI failures are due to this: https://github.com/JuliaLang/BinDeps.jl/issues/351

I would like to fix 0.7 deprecations and the above issue as well while I am at it. But without Compat.jl this is cumbersome. Any objections to adding the dependency back? BinDeps.jl installs it anyways.

**Comment by giordano** [comment]
I didn't have the time to have a look at this, just wanted to say that I think it's fine to require Compat

**Comment by helgee** [comment]
CI is green 🎉

File: src/erfa_common.jl [hunk 1]
```diff
@@ -1,66 +1,41 @@
-
```
Comment (giordano) [inline src/erfa_common.jl]: Looks like these constants are used nowhere, perhaps are better suited in the newly-born `AstroBase.jl` package.  And `DPI` and `D2PI` are probably redundant, given we have `pi` in Julia.  Anyway, this is unrelated to this change.

**Comment by helgee** [comment]
I will merge this later and tag a new release if I hear no objections.
